config/config.go

```diff
@@ -674,6 +674,11 @@ type P2PConfig struct { //nolint: maligned
 	// layer uses. Options are: "fifo" and "priority",
 	// with the default being "priority".
 	QueueType string `mapstructure:"queue-type"`
+
+	// UseLibP2P switches to using the new networking layer based
+	// on libp2p. This option is unlikely to persist into a
+	// release of tendermint, but will ease the transition.
+	UseLibP2P bool `mapstructure:"use-lib-p2p"`
```

Comment:
This might be a good case for an `experimental-*` prefix as a signal to the reader that it's going away by design.